packages/relay/src/lib/eth.ts

```diff
@@ -498,7 +499,7 @@ export class EthImpl implements Eth {
       // retrieve the contract result details
       await this.mirrorNodeClient.getContractResultsDetails(address, contractResult.results[0].timestamp)
         .then(contractResultDetails => {
-          if (contractResultDetails?.state_changes != null) {
+          if (contractResultDetails?.state_changes != null && Array.isArray(contractResultDetails?.state_changes) && contractResultDetails?.state_changes.length > 0) {
```

Comment:
`Array.isArray(null)` and `Array.isArray(undefined)` returns false so we can drop the initial null check
```suggestion
          if (Array.isArray(contractResultDetails?.state_changes) && contractResultDetails?.state_changes.length > 0) {
```

Reply:
(stupid mis-hit button ate my comment) the concept being expressed here "array must not be empty" is needlessly complicated when written out due to javascript/java/c# stupidity of allowing arrays to be "null" as a distinct thing to test syntactically than array being/not being empty.  So here, even with this simplification, it takes an entire line to express and a pattern match in your brain to figure out the idiom - again complicated because it is done w.r.t. to a named field of an object which itself takes 40 characters to express and you have to say it twice.  My suggestion would be to extract it into a named method - even though the method itself might be a one-liner - so that every _use_ is simplified and the name alone is sufficient to tell you what's going on - you don't have to recognize the pattern.  (Except of course the language might not actually allow you to easily express the method _call_.)

Reply:
I understand the sentiment.  With issues like this, I like to ask myself "does doing this lessen the likelihood of a mistake?".  If so then I create the helper static method.  I think one needs to be careful or you could spend so much time creating helper/macro methods, the code becomes difficult to follow (I've followed too many #define references in header files in c++ to find at the end something trivial).  I'll take your council in this case.